### PR TITLE
ci: replace set-output with new github actions syntax

### PR DESCRIPTION
**Pull request by cuyk**
## What
`set-output` GitHub Actions directive is deprecated and will be fully disabled on 31st May 2023.
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

## How
Replace `set-output` with new 

## 🚨 User Impact 🚨
No user impact

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: .github/actions/build-branch/action.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ inputs:
     required: false
 outputs:
   branch_version_tag:
-    description: 'Tag used for jars and docker images. Either user specified or auto generated as `dev-<commit_hash>`'
+    description: "Tag used for jars and docker images. Either user specified or auto generated as `dev-<commit_hash>`"
```
Comment (supertopher) [inline .github/actions/build-branch/action.yml:9]: is this for consistency, or was this is actual error?

File: .github/workflows/publish-helm-charts.yml [hunk 2]
```diff
@@ -135,4 +131,4 @@ jobs:
         run: |
           cd ./airbyte/
           git tag ${{ needs.generate-semantic-version.outputs.tag }}
-          git push origin ${{ needs.generate-semantic-version.outputs.tag }}
\ No newline at end of file
+          git push origin ${{ needs.generate-semantic-version.outputs.tag }}
```
Comment (supertopher) [inline .github/workflows/publish-helm-charts.yml:134]: <3

File: .github/workflows/publish-oss-for-cloud.yml [hunk 3]
```diff
@@ -61,16 +61,16 @@ jobs:
         shell: bash
         run: |-
           set -x
-          
+
```
Comment (supertopher) [inline .github/workflows/publish-oss-for-cloud.yml:64]: <3

**Review by git-phu — APPROVED** [review]
thanks!

**Review by xpuska513 — APPROVED** [review]
Looks good to me, approved.
Looks like they just added a new env var for storing outputs, guess that's just for mem optimisation

**Review by supertopher — APPROVED** [review]
This looks great @cuyk 

I think testing every pipeline seems extreme, but if you haven't already, can we test this pattern at least once either through manual work triggers or in the test actions repo?

File: .github/actions/build-branch/action.yml (hunk 1 above)
Comment (cuyk) [reply .github/actions/build-branch/action.yml:9]: it was pre-commit hook we have in the repo https://github.com/airbytehq/airbyte/blob/master/.pre-commit-config.yaml#L29

**Comment by cuyk** [comment]
> I think testing every pipeline seems extreme, but if you haven't already, can we test this pattern at least once either through manual work triggers or in the test actions repo?

I did that with another [draft pr](https://github.com/airbytehq/airbyte/actions/runs/3363164582/jobs/5575914019)

File: .github/actions/build-branch/action.yml (hunk 1 above)
Comment (supertopher) [reply .github/actions/build-branch/action.yml:9]: ty for context

**Pull request merged**